include/boost/beast/experimental/core/impl/url_parser.ipp

```diff
@@ -0,0 +1,352 @@
+//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#include <boost/beast/experimental/core/url_parser.hpp>
+
+namespace boost {
+namespace beast {
+
+//--------------------------------------------------------------------------
+// scheme:[//[user[:password]@]host[:port]][/path][?query][#fragment]
+
+using boost::beast::string_view;
+
+string_view url_parser::make_string_view(const char* b, std::size_t s)
+{
+    return string_view(b, s);
+}
+
+bool url_parser::isunreserved(const char c) const
+{
+    if (std::isalpha(c) || std::isdigit(c) ||
```

Comment:
These are slow, they will look at the current locale. Beast has its own private replacements which assume 7-bit ascii (the standard for HTTP and web). Examples:
https://github.com/boostorg/beast/blob/develop/include/boost/beast/core/string.hpp#L33
https://github.com/boostorg/beast/blob/develop/include/boost/beast/http/detail/rfc7230.hpp
https://github.com/boostorg/beast/blob/develop/include/boost/beast/http/detail/basic_parser.hpp#L55

Just to keep things interesting, note that if someone uses Beast and sets the current locale differently, then functions like `std::isdigit` can do the wrong thing with respect to RFC compliance. That is why we must avoid `std` char functions.

Reply:
> Actually, I have written a lot of these already for some partial work I did on my own URI parser you can copy them

Awesome, I'll add those as replacements.

I wasn't familiar with the other Beast implementations, thanks for the links.